drracket-tool-lib/drracket/private/standalone-module-browser.rkt

```diff
@@ -365,143 +363,137 @@
-        (send pasteboard render-snips)
-        
-        (send frame show #t)))))
+  (define success? (fill-pasteboard pasteboard filename show-status void))
+  (kill-thread thd)
+  (parameterize ([current-eventspace progress-eventspace])
+    (queue-callback (λ () (send progress-frame show #f))))
+  (when success?
+    (let ()
+      (define frame
+        (new overview-frame%
+             [label (string-constant module-browser)]
+             [width (preferences:get 'drracket:module-overview:window-width)]
+             [height (preferences:get 'drracket:module-overview:window-height)]
+             [alignment '(left center)]))
+      (define vp
+        (new vertical-panel%
+             [parent
+              (if (method-in-interface? 'get-area-container (object-interface frame))
+                  (send frame get-area-container)
+                  frame)]
+             [alignment '(left center)]))
+      (define root-message
+        (instantiate message% ()
+          [label (format (string-constant module-browser-root-filename) filename)]
+          [parent vp]
+          [stretchable-width #t]))
+      (define label-message
+        (instantiate message% ()
+          [label ""]
+          [parent vp]
+          [stretchable-width #t]))
+      (define font/label-panel (new horizontal-panel% [parent vp] [stretchable-height #f]))
+      (define pkg-choice
+        (new module-browser-pkg-set-choice% [parent font/label-panel] [pasteboard pasteboard]))
+      (define submod-choice
+        (new module-browser-submod-set-choice% [parent font/label-panel] [pasteboard pasteboard]))
+      (define font-size-gauge
+        (instantiate slider% ()
+          [label font-size-gauge-label]
+          [min-value 1]
+          [max-value 72]
+          [init-value (preferences:get 'drracket:module-overview:label-font-size)]
+          [parent font/label-panel]
+          [callback
+           (λ (x y) (send pasteboard set-label-font-size (send font-size-gauge get-value)))]))
+      (define module-browser-name-length-choice
+        (new choice%
+             (parent font/label-panel)
+             (label (string-constant module-browser-name-length))
+             (choices (list (string-constant module-browser-name-long)
+                            (string-constant module-browser-name-very-long)))
+             (selection (case (preferences:get 'drracket:module-browser:name-length)
+                          [(0) 0]
+                          [(1) 0]
+                          [(2) 0]
+                          [(3) 1]))
+             (callback
+              (λ (x y)
+                ;; note: the preference drracket:module-browser:name-length is also used for
+                ;; the View|Show Module Browser version of the module browser
+                ;; here we just treat any pref value except '3' as if it were for the long names.
+                (let ([selection (send module-browser-name-length-choice get-selection)])
```

Comment:
probably another one that didn't kick in, but it feels like it would be nice to limit the changes by region instead of by number of changes maybe?

Reply:
I'm considering redoing how the limiting works. It's hard to tell when something isn't being refactored because it hit the limit or because the refactoring rule didn't catch it.